core/src/test/scala/astraea/spark/rasterframes/TileStatsSpec.scala

```diff
@@ -62,6 +61,26 @@ class TileStatsSpec extends TestEnvironment with TestData {
         .as[(Int, Int)].first() === (3, 3))
     }
 
+    // tiles defined for the next few tests
+    val tile1 = fracTile(10, 10, 5)
+    val tile2 = ArrayTile(Array(-5, -4, -3, -2, -1, 0, 1, 2, 3), 3, 3)
+    val tile3 = randomTile(255, 255, IntCellType)
+    val ds = Seq[Tile](tile1, tile2, tile3).toDF("tiles")
+
+    it("should compute accurate item counts") {
+      val checkedValues = Seq[Double](0, 4, 7, 13, 26)
+      val result = checkedValues.map(x => ds.select(tileHistogram($"tiles")).first().itemCount(x))
+      checkedValues.foreach(x => assert((x == 0 && result.head == 4) || result.contains(x - 1)))
```

Comment:
Change `checkedValues.foreach` to `forEvery(checkedValues) { x => ... }`. The test framework gives you better error reports that way.